node/config/load.go

```diff
@@ -395,8 +398,32 @@ func ConfigUpdate(cfgCur, cfgDef interface{}, opts ...UpdateCfgOpt) ([]byte, err
 			return nil, xerrors.Errorf("parsing updated config: %w", err)
 		}
 
-		if !reflect.DeepEqual(cfgCur, cfgUpdated) {
-			return nil, xerrors.Errorf("updated config didn't match current config:")
+		opts := []cmp.Option{
+			// This equality function compares big.Int
+			cmpopts.IgnoreUnexported(big.Int{}),
+			cmp.Comparer(func(x, y []string) bool {
+				//both nil
+				if x == nil && y == nil {
+					return true
+				}
+				// both []string
+				if reflect.TypeOf(x) == reflect.TypeOf(y) {
```

Comment:
They both should always be []string even with nil and {}. (Though I don't know if it's guaranteed to be the same pointer). You just be able to presume you have a []string
but if you want to verify it, do this instead:

tx, ty := reflect.TypeOf(x), reflect.TypeOf(y)
if tx.Kind() == reflect.Array 
&& ty.Kind() == reflect.Array 
&& tx.Elem().Kind() == reflect.String 
&& ty.Elem().Kind() == reflect.String

All you really need to verify if that the contents match:

```return strings.Join(x, "\n") == strings.Join(y,"\n")```    // possibly sorted first

Reply:
I went with the type match. If both are Slice and elements are strings then we have a match.
Pushed the commit.